From ee6d19177876b7afd6c06b83d64e9f1124c2fe03 Mon Sep 17 00:00:00 2001
From: Piotr Marcinkowski <piotr.marcinkowski@redembedded.com>
Date: Fri, 10 Feb 2023 19:09:41 +0000
Subject: [PATCH] OMWAPPI-912 Fix artifacts when playing CBCS content

Currently subsamples are not being passed to OCDM as they are,
meaning they get changed by extracting only their encrypted part
to a separete buffer. That buffer, that consists only of combined
encrypted data of all samples is then passed to underlying
components - OCDM, SVP module, TEE. In order to indicate there
are no unencrypted bytes in that buffer the clear_bytes value
is set to 0 for each sub-sample.
---
 Source/ocdm/adapter/broadcom-svp/open_cdm_adapter.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/ocdm/adapter/broadcom-svp/open_cdm_adapter.cpp b/Source/ocdm/adapter/broadcom-svp/open_cdm_adapter.cpp
index f6cc2dc..aadcf1f 100644
--- a/Source/ocdm/adapter/broadcom-svp/open_cdm_adapter.cpp
+++ b/Source/ocdm/adapter/broadcom-svp/open_cdm_adapter.cpp
@@ -202,6 +202,7 @@ OpenCDMError opencdm_gstreamer_session_decrypt(struct OpenCDMSession* session, G
             subSampleInfoPtr = reinterpret_cast<SubSampleInfo*>(malloc(subSampleCount * sizeof(SubSampleInfo)));
             for (unsigned int position = 0; position < subSampleCount; position++) {
                 gst_byte_reader_get_uint16_be(reader, &subSampleInfoPtr[position].clear_bytes);
+                subSampleInfoPtr[position].clear_bytes = 0;
                 gst_byte_reader_get_uint32_be(reader, &subSampleInfoPtr[position].encrypted_bytes);
             }
             gst_byte_reader_set_pos(reader, 0);
-- 
2.25.1

